compiler/src/iree/compiler/Codegen/Common/GPU/GPUNestedLayoutDistributionPatterns.cpp

```diff
@@ -240,6 +240,36 @@ static VectorValue getSlicedPermutedMask(PatternRewriter &rewriter,
   return slicedMask;
 }
 
+/// Project a vector based on a provided projection map.
+/// Firstly, this will tranpose the vector in a way sliced out
+/// dims become outermost. Then it performs a vector.extract
+/// remove the dims that are not present in the results of the map.
+static VectorValue projectVector(RewriterBase &rewriter, Location loc,
```

Comment:
This is just a vector.extract_strided_slice no? I guess you are trying to project out the indexed dimensions, so this is okay. Just mention it in the comment.

Reply:
sure.. i ll add it to the comment.